service/README.md

```diff
@@ -64,6 +65,28 @@ class MyService2 implements MyContract2 {
 }
 ```
 
+Service with `java.util.function.Supplier` as a contract example:
+
+```java
+import java.util.function.Supplier;
+
+import io.helidon.service.registry.Service;
+
+@Service.Provider
+// the type must be fully qualified, as it is code generated
+class MyService3 implements Supplier<Optional<com.foo.bar.MyContract3>> {
+    private final MyContract dependency;
+
+    MyService3() {
+    }
+
+    @Override
+    public Optional<MyContract3> get() {
+         return "MyService3";
```

Comment:
This cannot compile; the returned value is not an instance of the declared return type.

Reply:
Good catch! I've update the example.